frontend/src/pages/PipelineDetails.tsx

```diff
@@ -293,8 +297,33 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
 
     this.props.updateToolbar({ breadcrumbs, actions: toolbarActions, pageTitle });
 
-    if (isFeatureEnabled(FeatureKey.V2) && isV2PipelineSpec(templateString)) {
-      const graphV2 = 'TO BE FULFILLED, non-empty string will open V2';
+    let graph: graphlib.Graph | null = null;
+    let reducedGraph: graphlib.Graph | null | undefined = null;
+    let graphV2: PipelineFlowElement[] = [];
+    if (templateString) {
+      try {
+        const template = JsYaml.safeLoad(templateString);
+        graph = StaticGraphParser.createGraph(template!);
+
+        reducedGraph = graph ? transitiveReduction(graph) : undefined;
+        if (graph && reducedGraph && compareGraphEdges(graph, reducedGraph)) {
+          reducedGraph = undefined; // disable reduction switch
+        }
+      } catch (v1err) {
+        try {
+          const pipelineSpec = convertJsonToV2PipelineSpec(templateString);
+          graphV2 = convertFlowElements(pipelineSpec);
+        } catch (v2err) {
+          if (isFeatureEnabled(FeatureKey.V2)) {
+            await this.showPageError('Error: failed to generate V2 Pipeline graph.', v2err);
+          } else {
+            await this.showPageError('Error: failed to generate V1 Pipeline graph.', v1err);
+          }
```

Comment:
nit: I believe we should only try to parse the template as V2 when V2 feature flag is enabled.

Reply:
Makes sense, fixed along with the above comment.